File: apps/betty_app/lib/trips/application/use_cases/get_current_trip_use_case.dart
import '../../infrastructure/services/trip_api_service.dart';
import '../../domain/entities/trip.dart';

class GetCurrentTripUseCase {
  final TripApiService _apiService;

  const GetCurrentTripUseCase(this._apiService);

  Future<Trip?> execute() async {
    return await _apiService.getCurrentTrip();
  }
}

File: apps/betty-server/src/trips/application/use-cases/get-current-trip/get-current-trip.use-case.ts
import { Injectable, Inject } from '@nestjs/common';
import { TripRepository } from '../../../domain/repositories/trip.repository';
import { Trip } from '../../../domain/entities/trip.entity';
import { TRIP_REPOSITORY } from '../../../infrastructure/ioc/symbols';

@Injectable()
export class GetCurrentTripUseCase {
  constructor(
    @Inject(TRIP_REPOSITORY)
    private readonly tripRepository: TripRepository
  ) {}

  async execute(): Promise<Trip | null> {
    return this.tripRepository.findCurrent();
  }
}
